### PR TITLE
Setup local mock json server to emulate API calls

### LINKED ISSUE #22: Setup JSON server to mock API / Data Response

### PR DESCRIPTION
**Pull request by nitishdhar**
- Adds `yarn api` and `yarn dev-api` scripts to run the API server
- Adds a playground page to add features and components when in development

### CODE REVIEW

File: app/playground/page.tsx [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { Metadata } from 'next';
+import PageClient from './page.client';
+
+export const metadata: Metadata = {
+  title: 'EDPN Playground',
+  description: 'Elite Dangerous Pilots Network',
+  icons: 'EDPN_logo.png',
```
Comment (aslink87) [inline app/playground/page.tsx]:
```suggestion
  icons: 'EDPN_logo_black.png',
```

This also needs updated in the FAQ page.tsx

File: app/playground/page.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+  return res.json();
+}
+
+export default async function Page() {
+  let posts = null;
+
+  try {
+    posts = await getPosts();
+  } catch (error) {
```
Comment (aslink87) [inline app/playground/page.tsx:53]: With the above !res.ok check would this catch ever throw an error? If not, should we consider adding an environment check to the getPosts  if check that consoles the error message in development?

File: package.json [hunk 3]
```diff
@@ -5,6 +5,8 @@
   "license": "MIT",
   "scripts": {
     "dev": "next dev",
+    "dev-api": "concurrently \"next dev\" \"yarn api\"",
```
Comment (aslink87) [inline package.json:8]: Should be added to README

File: app/playground/page.client.tsx [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            <p>Post Title: {item.title}</p>
+            <p>Author: {item.author}</p>
+          </div>
+        ))
+      ) : (
+        <Alert status="error">
+          <AlertIcon />
+          Failed to fetch post data
+        </Alert>
+      )}
```
Comment (aslink87) [inline app/playground/page.client.tsx:41]: Minor syntax point - should this ternary not be refactored with a [nullish coalescence operator?](https://github.com/airbnb/javascript#comparison--unneeded-ternary)

**Review by aslink87 — CHANGES_REQUESTED** [review]
I really like this. By putting it in an example page, you made it easy for me to understand what the intention of the feature is. Thanks

However, I don't see us making many GET requests. Most of our reqs will be POST with filters in the body. Even the system search component will have a body that includes the search string. That being said, does this json server provide the functionality to handle route logic like this? Or will we have to create api routes in NextJS to do this in conjunction with json server for the dev env?

At the end of the day, I think it will be better for the devs if this example included a POST req with some sort of filter function as that will be quite common in the app going forward.

File: app/playground/page.tsx (hunk 2 above)
Comment (nitishdhar) [reply app/playground/page.tsx:53]: This isn't representative of how we will make calls in production. I plan to do a [follow up PR](https://github.com/ed-pilots-network/frontend/issues/49) which will setup our API calling patterns e.g. switch to mock, switch to dev endpoint and even prod if needed.

The one here will not throw a breaking error, instead log it in the console of the server running at 3001.

File: app/playground/page.tsx (hunk 1 above)
Comment (nitishdhar) [reply app/playground/page.tsx]: Will update 👍

File: package.json (hunk 3 above)
Comment (nitishdhar) [reply package.json:8]: Will update 👍

File: app/playground/page.client.tsx (hunk 4 above)
Comment (nitishdhar) [reply app/playground/page.client.tsx:41]: Not sure how that can be applied here, I need to check for presence of posts and render different components based on that. Can you show an example? I mean I can avoid using ternary by making a function or something like this 

```
function Posts({posts}) {
  if(!posts) {
    return (
      <Alert status="error">
        <AlertIcon />
        Failed to fetch post data
      </Alert>
    );
  }

  return posts.map(post => <Post key={post.id} post={post} />);
}
```

File: app/playground/page.tsx (hunk 2 above)
Comment (aslink87) [reply app/playground/page.tsx:53]: Sounds good. We can chat more about that over there as I'd like to pick your brain more on it. What you've done here will be a nice launch-pad.

File: app/playground/page.client.tsx (hunk 4 above)
Comment (aslink87) [reply app/playground/page.client.tsx:41]:
```
{posts?.map((item) => (
  <div key={item.id}>
  <p>Post Title: {item.title}</p>
  <p>Author: {item.author}</p>
  </div>
)) ?? (
  <Alert status="error">
    <AlertIcon />
      Failed to fetch post data
    </Alert>
)}
```
isn't JSX pretty? lol

Comment (nitishdhar) [reply app/playground/page.client.tsx:41]: Damn, just posted a commit with everything else covered. Sending another commit

**Comment by nitishdhar** [comment]
Covered most feedback. Wrote a middleware based setup so we can make any other custom mods if needed in how the mock server should behave.

**Comment by aslink87** [comment]
lgtm. Being able to add an Express layer this is far more robust than I thought.

**Review by aslink87 — APPROVED** [review]

**Pull request merged**